test/metabase/test/data/materialize.clj

```diff
@@ -0,0 +1,105 @@
+   [metabase.test.data.interface :as tx]
+   [metabase.test.data.sql :as sql.tx]
+   [metabase.test.data.sql-jdbc.execute :as execute]
+   [metabase.test.data.sql-jdbc.load-data :as load-data]
+   [metabase.test.data.sql.ddl :as ddl]
+   [metabase.util.log :as log]))
+
+(set! *warn-on-reflection* true)
+
+(defmethod driver/database-supports? [:materialize :foreign-keys] [_driver _feature _db] (not config/is-test?))
+
+(defmethod ddl/drop-db-ddl-statements :materialize
+  [& args]
+  (apply (get-method ddl/drop-db-ddl-statements :sql-jdbc/test-extensions) args))
+
+(defmethod tx/aggregate-column-info :materialize
+  ([driver ag-type]
+   ((get-method tx/aggregate-column-info ::tx/test-extensions) driver ag-type))
+
+  ([driver ag-type field]
+   (merge
+    ((get-method tx/aggregate-column-info ::tx/test-extensions) driver ag-type field)
+    (when (= ag-type :sum)
+      {:base_type :type/BigInteger}))))
+
+(doseq [[base-type db-type] {:type/BigInteger     "BIGINT"
+                             :type/Boolean        "BOOL"
+                             :type/Date           "DATE"
+                             :type/DateTime       "TIMESTAMP"
+                             :type/Decimal        "DECIMAL"
+                             :type/Float          "FLOAT"
+                             :type/Integer        "INTEGER"
+                             :type/Text           "TEXT"
+                             :type/JSON           "JSON"
+                             :type/Time           "TIME"
+                             :type/UUID           "UUID"}]
+  (defmethod sql.tx/field-base-type->sql-type [:materialize base-type] [_ _] db-type))
+
+(defmethod tx/dbdef->connection-details :materialize
+  [_ context {:keys [database-name]}]
+  (merge
+   {:host     (tx/db-test-env-var-or-throw :materialize :host "localhost")
+    :port     (tx/db-test-env-var-or-throw :materialize :port 6875)
+    :ssl      (tx/db-test-env-var :materialize :ssl false)
+    :cluster  (tx/db-test-env-var :materialize :cluster "default")}
+   (when-let [user (tx/db-test-env-var :materialize :user)]
+     {:user user})
+   (when-let [password (tx/db-test-env-var :materialize :password)]
+     {:password password})
+   (when (= context :db)
+     {:db database-name})))
+
+(defmethod sql.tx/drop-table-if-exists-sql :materialize
+  [driver {:keys [database-name]} {:keys [table-name]}]
+  (format "DROP TABLE IF EXISTS \"%s\".\"%s\".\"%s\""
+          (ddl.i/format-name driver database-name)
+          "public"
+          (ddl.i/format-name driver table-name)))
+
+(defmethod sql.tx/create-db-sql :materialize
+  [driver {:keys [database-name]}]
+  (format "CREATE DATABASE \"%s\";" (ddl.i/format-name driver database-name)))
+(defmethod sql.tx/drop-db-if-exists-sql :materialize
+  [driver {:keys [database-name]}]
+  (format "DROP DATABASE IF EXISTS \"%s\";" (ddl.i/format-name driver database-name)))
+
+(defmethod sql.tx/add-fk-sql :materialize [& _] nil)
```

Comment:
What would happen if Metabase tries to add a FK in Materialize? Is that a possible scenario?

Reply:
The key part is this config in the driver itself:

```
(doseq [[feature supported?] {:foreign-keys              (not config/is-test?)
```

By default Metabase assumes that it needs foreign keys in order to do joins, but this is not the case for Materialize and some other DB drivers that they have. So we disable this only during the tests, as otherwise the Metatabase test suite tries to add FKs on the tables which we don't support.